plugin/pkg/scheduler/schedulercache/interface.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schedulercache
+
+import "k8s.io/kubernetes/pkg/api"
+
+// NodeInfoCache collects pods' information and provides node-level aggregated information.
+// It's intended to supplant system modeler in some cases for efficient lookup.
+// NodeInfoCache's operations are pod centric. It incrementally updates itself based on pod event.
+// Pod events are sent via network. We don't have guaranteed delivery of all events except last seen.
+// Thus, we organized the state machine flow of a pod's events and handle it clearly.
+//
+// State Machine (life cycle) of a pod in scheduler's cache:
+//
+//                                                +-------+
+//                                                |       |
+//                                                |       | Update
+//           Assume                Add            +       |
+// Initial +--------> Assumed +----------------> Added <--+
+//                      +                         +
+//                      |                         |
+//                      |                         |
+//                      | Remove                  | Remove
+//                      |                         v
+//                      +------------------->  Deleted
+//
+// Depending on the implementation, it might choose to manage the expiration of assumed pod.
+// Thus, a new state "Expired" could be added to the diagram:
+//
+//                                                     +--------+
+//                                                     |        |
+//                                                     |        | Update
+//                             Add                     +        |
+//                        +---------------+------->  Added <----+
+//                        |               |            +
+//                        |               |            |
+//                        |               |            |
+//           Assume       +    expire     +            |
+// Initial +-------> Assumed +--------> Expired        |
+//                        +               +            |Remove
+//                        |               |            |
+//                        |               |            v
+//                        +---------------+------> Deleted
+//                              Remove
```

Comment:
What's the difference between Assumed and Expired? It seems that Add and Remove works same from them